core/integration/module_typescript_test.go

```diff
@@ -641,6 +641,140 @@ func (ModuleSuite) TestTypescriptRuntimeDetection(ctx context.Context, t *testct
 	})
 }
 
+func (ModuleSuite) TestTypeScriptPackageManagerDetection(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	t.Run("should default to yarn", func(ctx context.Context, t *testctx.T) {
+		modGen := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work").
+			With(daggerExec("init", "--name=Package-Detection", "--sdk=typescript", "--source=."))
+
+		files, err := modGen.Directory(".").Entries(ctx)
+		require.NoError(t, err)
+		require.Contains(t, files, "yarn.lock")
+
+		// Check that the package manager is set to yarn.
+		packageManager, err := modGen.File("package.json").Contents(ctx)
+		require.NoError(t, err)
+		require.Contains(t, packageManager, `"packageManager": "yarn@1.22.22`)
+
+		// Verify that it executes dagger example properly.
+		out, err := modGen.With(daggerQuery(`{packageDetection{containerEcho(stringArg:"hello"){stdout}}}`)).Stdout(ctx)
+		require.NoError(t, err)
+		require.JSONEq(t, `{"packageDetection":{"containerEcho":{"stdout":"hello\n"}}}`, out)
+	})
+
+	t.Run("should use pnpm if set in package.json", func(ctx context.Context, t *testctx.T) {
+		modGen := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work").
+			With(daggerExec("init", "--name=Package-Detection", "--sdk=typescript", "--source=.")).
+			WithoutFile("yarn.lock").
+			WithoutFile("package.json").
+			WithNewFile("package.json", `
+{
+  "dependencies": {
+    "typescript": "^5.3.2",
+    "@dagger.io/dagger": "./sdk"
+  },
+	"packageManager": "pnpm@8.15.4"
+}`).
+			With(daggerExec("develop"))
+
```

Comment:
I do that: https://github.com/dagger/dagger/blob/0aefd1c51bda1bbfd78fac36c56c9e2dc8fc284b/sdk/typescript/runtime/main.go#L266

Reply:
Ok, then what I'm saying is that instead of:
1. dagger init 
2. change package.json
3. dagger develop
4. dagger call

You do:
1. Add custom package.json
2. dagger init
3. dagger call